app/src/main/java/com/mpieterse/stride/ui/layout/central/viewmodels/HabitViewerViewModel.kt

```
@@ -121,6 +151,82 @@ class HabitViewerViewModel @Inject constructor(
+        } catch (e: Exception) {
+            Clogger.e("HabitViewerViewModel", "Failed to update habit", e)
+            _state.update { it.copy(error = "Failed to update habit") }
+        }
+    }
+
+    private suspend fun fetchBitmap(url: String): Bitmap? = withContext(Dispatchers.IO) {
+        return@withContext try {
+            (URL(url).openConnection()).getInputStream().use { input ->
+                BitmapFactory.decodeStream(input)
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)